lib/src/Base/Stat/CovarianceModelImplementation.cxx

```diff
@@ -791,9 +791,63 @@ Bool CovarianceModelImplementation::isDiagonal() const
 /* Marginal accessor */
 CovarianceModel CovarianceModelImplementation::getMarginal(const UnsignedInteger index) const
 {
-  if (index >= outputDimension_) throw InvalidArgumentException(HERE) << "Error: index=" << index << " must be less than output dimension=" << outputDimension_;
-  if (outputDimension_ != 1) throw NotYetImplementedException(HERE) << "In CovarianceModelImplementation::getMarginal(const UnsignedInteger index) const";
-  return clone();
+  return getMarginal(Indices(1, index));
+}
+
+/* Marginal accessor */
+CovarianceModel CovarianceModelImplementation::getMarginal(const Indices & indices) const
+{
+  if (!indices.check(outputDimension_)) throw InvalidArgumentException(HERE) << "Error: index=" << indices << " must be less than output dimension=" << outputDimension_;
+  CovarianceModel result(clone());
+  if (outputDimension_ > 1)
+  {
+    const UnsignedInteger oldOutputDimension = outputDimension_;
+    result.getImplementation()->outputDimension_ = indices.getSize();
+    Point newAmplitude;
+    for (UnsignedInteger i = 0; i < indices.getSize(); ++ i)
+    {
+      newAmplitude.add(amplitude_[indices[i]]);
+    }
+    result.getImplementation()->amplitude_ = newAmplitude;
+    if (!isDiagonal())
+    {
+      CorrelationMatrix newCorrelationMatrix(indices.getSize());
+      for (UnsignedInteger i = 0; i < indices.getSize(); ++ i)
+      {
+        for (UnsignedInteger j = 0; j < i; ++ j)
+        {
+          newCorrelationMatrix(i, j) = result.getImplementation()->outputCorrelation_(indices[i], indices[j]);
+        }
+      }
+      result.getImplementation()->setOutputCorrelation(newCorrelationMatrix);
+    }
+
+    // strip out invalid active parameters indices
+    Indices activeParameters(result.getImplementation()->activeParameter_);
+    Indices newActiveParameters;
+    UnsignedInteger counter = 0;
+    for (UnsignedInteger i = 0; i < activeParameters.getSize(); ++ i)
+    {
+      if (activeParameters[i] < inputDimension_)
+      {
+        // scale
+        newActiveParameters.add(activeParameters[i]);
+      }
+      else if (activeParameters[i] < inputDimension_ + outputDimension_)
+      {
+        // amplitude (assumes ascending order)
+        newActiveParameters.add(inputDimension_ + counter);
+        ++ counter;
+      }
+      else
+      {
+        // others
+        newActiveParameters.add(activeParameters[i] - oldOutputDimension - indices.getSize());
+      }
```

Comment:
No. others might contain output correlation coefficients. In that case, the code here is wrong.

Reply:
Yes, I dont think we can resolve this correctly here, I'll rather just resize the activeParameters.

Reply:
ok